projects/templates/src/lib/components/po-page-dynamic-table/po-page-dynamic-table.component.ts

```diff
@@ -601,53 +605,94 @@ export class PoPageDynamicTableComponent extends PoPageDynamicListBaseComponent
 
     if (allow) {
       const uniqueKey = this.formatUniqueKey(item);
+      const resourceToRemoveKey = this.returnResourcesKeys([item]);
 
       if (typeof actionRemove === 'boolean' || newUrl) {
-        const isRemoveFromUI = true;
         return this.poPageDynamicService
           .deleteResource(uniqueKey, newUrl)
-          .pipe(map(() => this.removeFromUI(item, isRemoveFromUI)));
+          .pipe(map(() => this.removeFromUI(resourceToRemoveKey, this.literals.removeSuccessNotification)));
       }
-      return of(actionRemove(uniqueKey, item)).pipe(map(remove => this.removeFromUI(item, remove)));
+
+      return of(actionRemove(uniqueKey, item)).pipe(
+        tap(remove => {
+          const removeItem = remove ?? false;
+          this.removeFromUI(resourceToRemoveKey, this.literals.removeSuccessNotification, removeItem);
+        })
+      );
     }
 
     return of({});
   }
 
-  private removeFromUI(item, remove) {
-    if (remove === true) {
-      this.removeLocalItems([item]);
-      this.poNotification.success(this.literals.removeSuccessNotification);
+  private removeFromUI(items: Array<any>, message: string, remove = true) {
+    if (remove === true && items?.length) {
+      this.removeLocalItems(items);
+      this.poNotification.success(message);
     }
   }
 
-  private removeAll() {
-    // TODO: usar propriedade nova pra validar os itens selecionados
-    const selectedItems = this.items.filter(item => item.$selected);
+  private removeAll(
+    actionRemoveAll: PoPageDynamicTableActions['removeAll'],
+    actionBeforeRemoveAll: PoPageDynamicTableActions['beforeRemoveAll']
+  ) {
+    const originalResourcesKeys = this.getSelectedItensKeysToRemove();
```

Comment:
Alterar `getSelectedItensKeysToRemove` para `getSelectedItemsKeysToRemove`:
```suggestion
    const originalResourcesKeys = this.getSelectedItemsKeysToRemove();
```